src/full/Agda/TypeChecking/DeadCode.hs

```diff
@@ -67,9 +66,7 @@ eliminateDeadCode !scope = Bench.billTo [Bench.DeadCode] $ do
   let !rootExtraDefs = mapMaybe extraRootsFilter $ HMap.toList defs
   let !rootRewrites  = sig ^. sigRewriteRules
 
-  -- Andreas, Oskar, 2023-10-19, issue #6931:
-  -- Needed to avoid deleting parameter sections of empty modules.
-  let !rootModSections = (sig ^. sigSections) `Map.intersection` pubModules
+  let !rootModSections = sig ^. sigSections
```

Comment:
@AndrasKovacs : Would be good to add a reference to issue #7382 in the code, for future reference and code comprehension.

Also please keep the reference to #6931 in.

Reply:
Uhh, sorry for the hasty merge.

I deleted the old comment intentionally because it would be confusing for new code reviewers and it's not relevant now. The empty modules are a red herring, the previous implementation could generated ill-formed interfaces all the same, it's just that we haven't run into errors because of it.

Reply:
I think referencing these two issues is still information a future developer can use to get more information about the topic.